it/cloud-search-index-storage.test.js

```diff
@@ -0,0 +1,298 @@
+    try {
+      const contentRecord1 = {
+        contentHash,
+        sourceName: 'file1.png',
+      };
+      const contentRecord2 = {
+        contentHash,
+        sourceName: 'file2.png',
+      };
+      const saveResult1 = await searchIndexStorage.save(contentRecord1);
+      const saveResult2 = await searchIndexStorage.save(contentRecord2);
+      assert.notEqual(undefined, saveResult1);
+      assert.notEqual(undefined, saveResult2);
+      await waitForRecord(10000);
+
+      await searchIndexStorage.updateByContentHash({ sourceType: 'test', contentHash });
+      await waitForRecord(10000);
+
+      const objectIDs = await searchIndexStorage.getObjectIdsByContentHash(contentHash);
+      assert.strictEqual(objectIDs.length, 2);
+      for (const objectID of objectIDs) {
+        // eslint-disable-next-line no-await-in-loop
+        const getResult = await searchIndexStorage.get(objectID);
+        assert.strictEqual(getResult.hits[0].sourceType, 'test');
+        assert.strictEqual(getResult.hits[0].contentHash, contentHash);
+      }
+    } finally {
+      await searchIndexStorage.deleteByContentHash(contentHash);
+    }
+  }).timeout(25000);
+
+  it('Update a content record', async () => {
+    let saveResult;
+    try {
+      const contentRecord = generateContentRecord();
+      saveResult = await searchIndexStorage.save(contentRecord);
+      assert.notEqual(undefined, saveResult);
+      await waitForRecord(10000);
+
+      await searchIndexStorage.update([{ objectID: saveResult.objectID, sourceType: 'test' }]);
+      await waitForRecord(10000);
+
+      assert.ok(saveResult?.objectID);
+      const { objectID } = saveResult;
+      const getResult = await searchIndexStorage.get(objectID);
+      assert.ok(getResult);
+      assert.strictEqual(getResult.hits[0].sourceType, 'test');
+    } finally {
+      if (saveResult?.objectID) {
+        await searchIndexStorage.delete(saveResult.objectID);
+      }
+    }
+  }).timeout(25000);
+
+  it('Only update a content record with allowed fields', async () => {
+    let saveResult;
+    try {
+      const contentRecord = generateContentRecord();
+      saveResult = await searchIndexStorage.save(contentRecord);
+      assert.notEqual(undefined, saveResult);
+      await waitForRecord(10000);
+
+      await searchIndexStorage.update([{ objectID: saveResult.objectID, sourceType: 'test', newField: 'test' }]);
+      await waitForRecord(10000);
+
+      assert.ok(saveResult?.objectID);
+      const { objectID } = saveResult;
+      const getResult = await searchIndexStorage.get(objectID);
+      assert.ok(getResult);
+      assert.strictEqual(getResult.hits[0].sourceType, 'test');
+      assert.strictEqual(getResult.hits[0].newField, undefined);
+    } finally {
+      if (saveResult?.objectID) {
+        await searchIndexStorage.delete(saveResult.objectID);
+      }
+    }
+  }).timeout(25000);
+
+  it('Delete all objects of a source type', async () => {
```

Comment:
Can you please add tests to cover the other two delete methods (delete by contentHash, delete by objectID)?